client/qiskit_serverless/core/function.py

```diff
@@ -91,6 +96,70 @@ def __str__(self):
     def __repr__(self):
         return self.__str__()
 
+    def _validate_function(self) -> Tuple[bool, List[str]]:
+        """Validate function arguments using schema provided.
+
+        Returns:
+            Tuple[bool, List[str]]:
+                boolean specifiying if function arguments are valid
+                list of validation errors, if any
+        """
+        return True, []
+
+
+class RunService(ABC):
+    """Provide access to run a function and retrieve the jobs associated to that function"""
+
+    @abstractmethod
+    def jobs(self, **kwargs) -> List[Job]:
+        """Return list of jobs.
+
+        Returns:
+            list of jobs.
+        """
+
+    @abstractmethod
+    def run(
+        self,
+        program: Union[QiskitFunction, str],
+        arguments: Optional[Dict[str, Any]] = None,
+        config: Optional[Configuration] = None,
+    ) -> Job:
+        """Run a function and return its job."""
+
+
+class RunnableQiskitFunction(QiskitFunction):
+    """Serverless QiskitPattern.
+
+    Args:
+        title: program name
+        provider: Qiskit Function provider reference
+        entrypoint: is a script that will be executed as a job
+            ex: job.py
+        env_vars: env vars
+        dependencies: list of python dependencies to execute a program
+        working_dir: directory where entrypoint file is located (max size 50MB)
+        description: description of a program
+        version: version of a program
+    """
+
+    _run_service: RunService = None
```

Comment:
```suggestion
    _run_service: Optional[RunService] = None
```

Reply:
In this case i think it shouldnt be optional because, to be runnable it needs an associated run service. If run service will be none, then it is just a `QiskitFunction`